frontend/src/lib/StaticGraphParser.ts

```diff
@@ -263,3 +263,51 @@ export function createGraph(workflow: Workflow): dagre.graphlib.Graph {
 
   return graph;
 }
+
+/**
+ * Perform a transitive reduction over the input graph.
+ *
+ * From [1]: Transitive reduction of a directed graph D is another directed
+ * graph with the same vertices and as few edges as possible, such that for all
+ * pairs of vertices v, w a (directed) path from v to w in D exists if and only
+ * if such a path exists in the reduction
+ *
+ * [1]: https://en.wikipedia.org/wiki/Transitive_reduction
+ *
+ * @param graph The dagre graph object
+ */
+export function transitiveReduction(graph: dagre.graphlib.Graph) {
+  let visited: string[] = [];
+  const dfs_with_removal = (current: string, parent: string) => {
+    graph.successors(current)?.forEach((node: any) => {
+      if (visited.includes(node)) return;
+      visited.push(node);
+      if (graph.successors(parent)?.includes(node)) {
+        graph.removeEdge(parent, node);
+      }
+      dfs_with_removal(node, parent);
+    });
+  };
+
+  graph.nodes().forEach(node => {
+    visited = []; // clean this up before each new DFS
+    // start a DFS from each successor of `node`
+    graph.successors(node)?.forEach((successor: any) => dfs_with_removal(successor, node));
+  });
+  return graph;
+}
+
+export function compareGraphEdges(graph1: dagre.graphlib.Graph, graph2: dagre.graphlib.Graph) {
```

Comment:
I'm not seeing why this is useful.
I think hiding the sth dynamically in the UI is generally a bad design, users will be confused, because when the switch doesn't show up. There's no clue why it doesn't.

In this specific case, I think the simplest design is enough:
* always show the simplify graph switch

Reply:
So this is used both in RunDetails and PipelineDetails pages to compare the reduced graph vs the original one. If there is a difference then the reduced one gets picked to be displayed.